### PR TITLE
Fix the configuration of `Aggregation`  and `GC Count` metrics for oap self observability(#8707)

### LINKED ISSUE #8707: [Bug] the `Aggregation` and `GC Count` metrics in `fetcher-prom-rules/self.yaml` and `otel-oc-rules/oap.yaml` are wrong
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

OAP server (apache/skywalking)

### What happened

```
 - name: instance_metrics_first_aggregation
    exp: metrics_aggregation.tagEqual('dimensionality', 'minute', 'level', '1').sum(['service', 'instance']).increase('PT5M')
  - name: instance_metrics_second_aggregation
    exp: metrics_aggregation.tagEqual('dimensionality', 'minute', 'level', '2').sum(['service', 'instance']).increase('PT5M')
```
 
The value of first and second aggregation are wrong.

### What you expected to happen

Both of first and second aggregation use the metrics_aggregation metric.  the increase method uses <code>org.apache.skywalking.oap.meter.analyzer.dsl.counter.CounterWindow</code> to calculate.

The CounterWindow class is a Singleton class.  it uses <code>Map</code> to store the window data. the Map key is combined by name and labels of a metric. but both of  first and second aggregation have the same metric name and labels. the metric name is "metrics_aggregation" and the labels are "service" and "instance".  this will result in that they will use the same window.

the increase method is as below

``` java
  public Tuple2<Long, Double> increase(String name, ImmutableMap<String, String> labels, Double value, long windowSize, long now) {
        ID id = new ID(name, labels);
        Queue<Tuple2<Long, Double>> window = windows.computeIfAbsent(id, unused -> new PriorityQueue<>());
        window.offer(Tuple.of(now, value));
        long waterLevel = now - windowSize;
        Tuple2<Long, Double> peek = window.peek();
        if (peek._1 > waterLevel) {
            return peek;
        }

        Tuple2<Long, Double> result = peek;
        while (peek._1 < waterLevel) {
            result = window.poll();
            peek = window.element();
        }

        // Choose the closed slot to the expected timestamp
        if (waterLevel - result._1 <= peek._1 - waterLevel) {
            return result;
        }

        return peek;
    }
```

The right configurations i think and have proved are as below.

- self.yaml in fetcher-prom-rules

> I delete the first and second aggregation and create a new metric named "instance_metrics_aggregation".

```
 - name: instance_metrics_aggregation
    exp: metrics_aggregation.tagEqual('dimensionality', 'minute').sum(['service', 'instance', 'level']).increase('PT5M').tag({tags -> if (tags['level'] == '1') {tags.level = 'first aggregation'} }).tag({tags -> if (tags['level'] == '2') {tags.level = 'second aggregation'} })

```

- self-observability in ui-initialized-templates

```
      {
        "width": "3",
        "title": "Aggregation",
        "height": "200",
        "entityType": "ServiceInstance",
        "independentSelector": false,
        "metricType": "LABELED_VALUE",
        "metricName": "meter_oap_instance_metrics_aggregation",
        "queryMetricType": "readLabeledMetricsValues",
        "chartType": "ChartBar",
        "unit": "Per 5 Minute"
      }
```

### How to reproduce

I have debuged in the increase method and watch the metric value in the ui page.

### Anything else

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Cool-Coding**
### Fix <bug description or the bug issue number or bug issue link>
- [x] Explain briefly why the bug exists and how to fix it.

- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #8707 .
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/CHANGES.md).

**The Reason:**

The sum method loses the dimension label. When there are multiple metrics generated from the same child metric using increase method after sum. the value of the metrics will be miscalculated.

**How to Fix:**
- Delete the multiple metrics that generate from the same child metric
- Create a new metric, add the dimension label in sum method, change the tag of the new metric if needed.
- Fetch the new metric in the way of labeled value type.

### CODE REVIEW

**Comment by wu-sheng** [comment]
@Cool-Coding It seems you forget to change e2e-so11y. You changed the metrics type, please fix them.

File: test/e2e-v2/cases/so11y/so11y-cases.yaml [hunk 1]
```diff
@@ -30,7 +30,7 @@
       expected: expected/metrics-has-value.yml
     - query: swctl --display yaml --base-url=http://${oap_host}:${oap_12800}/graphql metrics linear --name=meter_oap_instance_trace_count --instance-name=http://localhost:1234 --service-name=oap::oap-server |yq e 'to_entries' -
       expected: expected/metrics-has-value.yml
-    - query: swctl --display yaml --base-url=http://${oap_host}:${oap_12800}/graphql metrics linear --name=meter_oap_instance_metrics_first_aggregation --instance-name=http://localhost:1234 --service-name=oap::oap-server |yq e 'to_entries' -
+    - query: swctl --display yaml --base-url=http://${oap_host}:${oap_12800}/graphql metrics linear --name=meter_oap_instance_metrics_aggregation --instance-name=http://localhost:1234 --service-name=oap::oap-server |yq e 'to_entries' -
```
Comment (wu-sheng) [inline test/e2e-v2/cases/so11y/so11y-cases.yaml:33]: You are changing `single value` -> `labeled value`, I assume the expected file should be updated too.

Comment (Cool-Coding) [reply test/e2e-v2/cases/so11y/so11y-cases.yaml:33]: > You are changing `single value` -> `labeled value`, I assume the expected file should be updated too.

I am not familiar  with `e2e-solly`. Take a moment, I will look at it. Thanks for your reminder and patience.

Comment (wu-sheng) [reply test/e2e-v2/cases/so11y/so11y-cases.yaml:33]: You could copy this expected file into the so11y folder, https://github.com/apache/skywalking/blob/master/test/e2e-v2/cases/vm/expected/metrics-has-value-label.yml

Comment (Cool-Coding) [reply test/e2e-v2/cases/so11y/so11y-cases.yaml:33]: > You could copy this expected file into the so11y folder, https://github.com/apache/skywalking/blob/master/test/e2e-v2/cases/vm/expected/metrics-has-value-label.yml

ok

Comment (Cool-Coding) [reply test/e2e-v2/cases/so11y/so11y-cases.yaml]: @wu-sheng I have run the "swctl" command in my own computer. please help me review again. Thanks!

File: test/e2e-v2/cases/so11y/so11y-cases.yaml [hunk 2]
```diff
@@ -30,8 +30,8 @@
       expected: expected/metrics-has-value.yml
     - query: swctl --display yaml --base-url=http://${oap_host}:${oap_12800}/graphql metrics linear --name=meter_oap_instance_trace_count --instance-name=http://localhost:1234 --service-name=oap::oap-server |yq e 'to_entries' -
       expected: expected/metrics-has-value.yml
-    - query: swctl --display yaml --base-url=http://${oap_host}:${oap_12800}/graphql metrics linear --name=meter_oap_instance_metrics_first_aggregation --instance-name=http://localhost:1234 --service-name=oap::oap-server |yq e 'to_entries' -
-      expected: expected/metrics-has-value.yml
+    - query: swctl --display yaml --base-url=http://${oap_host}:${oap_12800}/graphql metrics multiple-linear --name=meter_oap_instance_metrics_aggregation --labels=level --instance-name=http://localhost:1234 --service-name=oap::oap-server |yq e 'to_entries | with(.[] ; .value=(.value | to_entries))' -
+      expected: expected/metrics-has-value-label.yml
```
Comment (wu-sheng) [inline test/e2e-v2/cases/so11y/so11y-cases.yaml:34]: @wankai123 Could you check this? It seems tests are still failure.

File: test/e2e-v2/cases/so11y/so11y-cases.yaml [hunk 3]
```diff
@@ -30,8 +30,8 @@
       expected: expected/metrics-has-value.yml
     - query: swctl --display yaml --base-url=http://${oap_host}:${oap_12800}/graphql metrics linear --name=meter_oap_instance_trace_count --instance-name=http://localhost:1234 --service-name=oap::oap-server |yq e 'to_entries' -
       expected: expected/metrics-has-value.yml
-    - query: swctl --display yaml --base-url=http://${oap_host}:${oap_12800}/graphql metrics linear --name=meter_oap_instance_metrics_first_aggregation --instance-name=http://localhost:1234 --service-name=oap::oap-server |yq e 'to_entries' -
-      expected: expected/metrics-has-value.yml
+    - query: swctl --display yaml --base-url=http://${oap_host}:${oap_12800}/graphql metrics multiple-linear --name=meter_oap_instance_metrics_aggregation --labels=level --instance-name=http://localhost:1234 --service-name=oap::oap-server |yq e 'to_entries | with(.[] ; .value=(.value | to_entries))' -
```
[the remark this answers is not in the document]
Comment (wankai123) [reply test/e2e-v2/cases/so11y/so11y-cases.yaml:33]: The `--labels=` needs to provide the label-value .

**Review by wankai123 — CHANGES_REQUESTED** [review]

File: test/e2e-v2/cases/so11y/so11y-cases.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (Cool-Coding) [reply test/e2e-v2/cases/so11y/so11y-cases.yaml]: > The `--labels=` needs to provide the label-value .

@wankai123 Thank you very much!

**Review by wu-sheng — APPROVED** [review]

**Review by wankai123 — APPROVED** [review]

**Pull request merged**